dali/pipeline/operators/util/pad.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_UTIL_PAD_H_
+#define DALI_PIPELINE_OPERATORS_UTIL_PAD_H_
+
+#include <cstring>
+#include <utility>
+#include <vector>
+
+#include "dali/core/common.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/pipeline/operators/operator.h"
+#include "dali/kernels/common/pad.h"
+#include "dali/kernels/scratch.h"
+
+namespace dali {
+
+template <typename Backend>
+class Pad : public Operator<Backend> {
+ public:
+  inline explicit Pad(const OpSpec &spec)
+    : Operator<Backend>(spec),
+    padding_value_(spec.GetArgument<float>("padding_value")),
+    axis_(spec.GetArgument<int>("axis")) {
+  }
+
+ protected:
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
+    return false;
+  }
+
+  void RunImpl(Workspace<Backend> *ws) override;
+
+
+  USE_OPERATOR_MEMBERS();
+  using Operator<Backend>::RunImpl;
+
+  float padding_value_;
+  int axis_;
+  // In current implementation scratchpad memory is only used in the GPU kernel
+  // In case of using scratchpad in the CPU kernel a scratchpad allocator per thread
+  // should be instantiated
+  std::conditional_t<std::is_same<Backend, GPUBackend>::value,
```

Comment:
Consider using `KernelManager` to simplify the logic of your operators (see resize_base.*)

Reply:
Done